### PR TITLE
Bug 1244108 - [Messages][Tests] Fix and enable integration test for the Conversation-to-Participants path while in activity mode

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1883c9877b4a88fe7348a0c7c6bf7d662b3e49ae

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=482d727760af6100af1feb8ac0b1be18b3d531ab

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7eb744b20925c2092175117a1e866459acd29d7c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5bf0fcd26d5ad385be0e73f6443fd525c08e3796

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cafa5902e6f52d974968dd5491dd221ad0a1ec9a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=34ec3175c6a159e1232686c053665b23f1163168

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=eda883b3c6bc56f25838b45578e34a7fc9ca4f8e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a5fcecafa33c2fd678466f34dc127d782ddd7bb0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6d5c70f6cf4ee41235f7782d374bc99fc00c2e73

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9944fd0cd04794cfc6696fb9791e83d05821a90c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3b9b4dfe010fe4ffa2535bdd4a2cebda498ebc4c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f16bf731e168f7c9f5f22b5b3bc7ff236c026847

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fc0c4d5f2a0b67b4b40c4645614ba9011c199d04

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1f8abbd1762cafa70c9539da2796a257dda82c71

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a2fc25eb78f85e6563b3061b65222a2ae07adb31

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a198decdb2c224b1a62176dcb8b307fb95d5543d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=10193639c0cb5818b48a8de21cd26fbd14e336a3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4eff065033a840adaaa9585e02b7705a7ead70f0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1b381d5fbf9cfa8fcf9cec2cf3d579b0b1f1d73b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=579abd997d57236028a00eadbb302cf6fb130506

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d4a07093fdc9d8c92976402df8379eeb8d7392b8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1b12cae0463ce31318a070553aa5682ce3ec4867

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a5fa3d551c6d3c0a7025079af71704185ab05d42

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=516b403a8207036083842d14df654adc40dbc847

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3326f4b71febef3a5e37a015805b953b3de8dda5

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3d8f885174b0c3d85c6f132d0ebe769bb125c57e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=48c0c6a9bc462341e2c0a5a0977f61e5b5c484af

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=50612338aecd91bc468cdcba3c20c59701924acb

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9cb4c879336ceb573cf891b8fa77eeefbdbd5cd2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=780ef232357ffbcb3c05bfa7fcbdd27e9d2d8a50

File: apps/sms/test/marionette/lib/views/conversation/accessors.js [hunk 1]
```diff
@@ -122,9 +122,15 @@ ConversationAccessor.prototype = {
 
   waitToAppear: function() {
     var conversationPanel = this.client.helper.waitForElement(SELECTORS.main);
-    this.client.waitFor(function() {
-      return conversationPanel.location().x === 0;
+
+    this.client.waitFor(() => {
+      return conversationPanel.location().x === 0 && this.client.executeScript(
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: At first I was thinking about using smth like ```getActivePanelName``` from main messages lib, but it seems it doesn't distinguish conversation and composer.

Also I was considering moving this ```executeScript``` part to shared/app_accessors (that doesn't exist yet) that will expose such method with ```panelName``` parameter - but my rule of thumb is not to generalize unless we have more than one usage place :) But tell me what you think!

File: apps/sms/test/marionette/lib/views/new-message/view.js [hunk 2]
```diff
@@ -144,10 +148,14 @@ NewMessageView.prototype = {
     return conversationView;
   },
 
-  backToInbox: function() {
+  back: function() {
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/new-message/view.js]: Not sure if it's the best name, but we have similar method pair in conversation - so kept the same principle.

File: apps/sms/test/marionette/share_activity_test.js [hunk 3]
```diff
@@ -33,105 +34,108 @@ marionette('Messages as share target', function() {
...
         );
 
-        messagesApp.send();
-
         // Go to Participants panel
-        messagesApp.Conversation.headerTitle.tap();
-        messagesApp.Participants.waitToAppear();
+        var conversationView = newMessageView.send();
+
+        assert.equal(conversationView.headerTitle, '+1\n (+1)');
```
Comment (azasypkin) [inline apps/sms/test/marionette/share_activity_test.js]: Grr, I'll remove this assert if we land this PR - just forgot to remove :/ (Though I was quite surprised that we have ```\n``` here :))

File: apps/sms/test/marionette/lib/views/conversation/accessors.js (hunk 1 above)
Comment (julienw) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: nowadays location() is a synonym to rect() except location() is obsolete. So you can switch to rect().

Comment (julienw) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: wondering if the whole waitToAppear could be generalized actually...

File: apps/sms/test/marionette/share_activity_test.js [hunk 4]
```diff
@@ -33,105 +34,108 @@ marionette('Messages as share target', function() {
 
   suite('Share via Messages', function() {
     suite('Activity close button', function() {
+      var newMessageView;
```
Comment (julienw) [inline apps/sms/test/marionette/share_activity_test.js]: nit: use let :)

File: apps/sms/test/marionette/share_activity_test.js [hunk 5]
```diff
@@ -5,6 +5,7 @@ var assert = require('assert');
 
 var Messages = require('./lib/messages.js');
 var MessagesActivityCaller = require('./lib/messages_activity_caller.js');
+var NewMessageView = require('./lib/views/new-message/view');
```
Comment (julienw) [inline apps/sms/test/marionette/share_activity_test.js]: nit: use const :)

File: apps/sms/test/marionette/share_activity_test.js [hunk 6]
```diff
@@ -33,105 +34,108 @@ marionette('Messages as share target', function() {
...
-        messagesApp.send();
+        newMessageView.addNewRecipient('+1');
+        var conversationView = newMessageView.send();
 
-        // Go to the Report panel
-        messagesApp.contextMenu(messagesApp.Conversation.message);
-        messagesApp.Menu.selectAppMenuOption('View message report');
-        messagesApp.Report.waitToAppear();
+        var reportView = conversationView.openReport(
+          conversationView.messages()[0].id
```
Comment (julienw) [inline apps/sms/test/marionette/share_activity_test.js]: nit: `messages({ first: 1 })`

File: apps/sms/test/marionette/lib/views/conversation/accessors.js (hunk 1 above)
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: Likely yes, kind of ```waitForViewToAppear(viewElement)``` somewhere in common place. Please, tell me if you want me to do this in this pr

File: apps/sms/test/marionette/share_activity_test.js (hunk 6 above)
Comment (azasypkin) [inline apps/sms/test/marionette/share_activity_test.js]: There is only one anyway :) But yeah, maybe it's better for clarity

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=915179b49113a3831968e5d1fd5a921a9f69eca3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2f67e966311f54a37c71e93a292e62db9ed68a1d

**Pull request merged**